yt/data_objects/construction_data_containers.py

```diff
@@ -899,6 +913,36 @@ def _fill_fields(self, fields):
             fi = self.ds._get_field_info(field)
             self[field] = self.ds.arr(dest, fi.units)
 
+    def _fill_sph_particles(self, fields):
+        # checks that we have the field and gets information
+        fields = [f for f in fields if f not in self.field_data]
+        if len(fields) == 0: return
+
+        ptype = self.ds._sph_ptype
+        for field in fields:
+            dest = np.zeros(self.ActiveDimensions, dtype="float64")
+
+            for chunk in self._data_source.chunks([field], "io"):
+                px = chunk[(ptype,'particle_position_x')].in_base("code")
+                py = chunk[(ptype,'particle_position_y')].in_base("code")
+                pz = chunk[(ptype,'particle_position_z')].in_base("code")
+                hsml = chunk[(ptype,'smoothing_length')].in_base("code")
+                mass = chunk[(ptype,'particle_mass')].in_base("code")
+                dens = chunk[(ptype,'density')].in_base("code")
+                field_quantity = chunk[(ptype, field[1])].in_base("code")
```

Comment:
It turns out if you ask for temperature, this dies for gadget data, because gadget doesn't define `code_temperature' units. In principle this should work but we probably need fixes elsewhere. For now I think we can drop `.in_base('code')` on this one line

Reply:
ah cool! I guess that is fine for now if you are planning to update the field system at some point